testbeam_analysis/tools/kalman.py

```diff
@@ -152,12 +158,63 @@ def _filter(transition_matrices, observation_matrices, transition_covariances,
     kalman_gains = np.zeros((chunk_size, n_timesteps, n_dim_state, n_dim_obs))
     filtered_states = np.zeros((chunk_size, n_timesteps, n_dim_state))
     filtered_state_covariances = np.zeros((chunk_size, n_timesteps, n_dim_state, n_dim_state))
+    # array where new transition matrices are stored, needed to pass it to kalman smoother
+    transition_matrices_update = np.zeros_like(transition_covariances)
 
     for t in range(n_timesteps):
         if t == 0:
             predicted_states[:, t] = initial_state
```

Comment:
This is done [here](https://github.com/YannickDieter/testbeam_analysis/blob/development/testbeam_analysis/tools/kalman.py#L170-L215).